### DISCUSSION TITLE
VSCode being a bitch about HScript

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#405] [Q&A] [answered]
I think that VSCode is too busy telling me that `[event name]_Fields_ isn't valid` to actually help with HScripting. I'm trying to port this psychlua event to hscript and I'm wondering how I should go about putting this piece of GUI on the hudcamera?
Code I'm making:
![image](https://github.com/user-attachments/assets/f35ee6a3-1056-4c03-8a2a-4b23fa9f902d)
PsychLUA:
![image](https://github.com/user-attachments/assets/01dad048-88a1-440a-8443-c113b8fbfa07)
Also here's what I mean by VSCode complaining:
![image](https://github.com/user-attachments/assets/aef9f1f3-e018-47f6-aeff-d13a1f4022ac)

Wondering if there's even a Codename-HScript VSCode plugin? I just need help in general.

**Comment by NexIsDumb** [comment, marked as answer]
this is literally impossible to make work lmfao, itd have to compile the main source of cne first like it happens when u edit the cne source itself
although you can disable the errors if you place a comment on the first line like we do already in some cne scripts; for example:

```haxe
//
import shits.etc.etc;
```

**Reply by THEkatinamicrowave** [reply]
thanks that works for me!

**Reply by swordcube** [reply]
> this is literally impossible to make work lmfao, itd have to compile the main source of cne first like it happens when u edit the cne source itself although you can disable the errors if you place a comment on the first line like we do already in some cne scripts; for example:
> 
> ```hx
> //
> import shits.etc.etc;
> ```

you can also disable the errors by using .hxs files, and then configuring vscode to highlight haxe for it (the errors only happen with .hx files):
![image](https://github.com/user-attachments/assets/a976b56c-a50a-4c93-b213-62cf4bc6afbd)
![image](https://github.com/user-attachments/assets/ca87fa01-e4ad-43e0-8d9f-016dd22496c9)
![image](https://github.com/user-attachments/assets/304062df-0138-422a-b256-b81606648364)

**Reply by NexIsDumb** [reply]
> > this is literally impossible to make work lmfao, itd have to compile the main source of cne first like it happens when u edit the cne source itself although you can disable the errors if you place a comment on the first line like we do already in some cne scripts; for example:
> > 
> > ```hx
> > //
> > import shits.etc.etc;
> > ```
> 
> you can also disable the errors by using .hxs files, and then configuring vscode to highlight haxe for it (the errors only happen with .hx files):
> ![image](https://github.com/user-attachments/assets/a976b56c-a50a-4c93-b213-62cf4bc6afbd)
> ![image](https://github.com/user-attachments/assets/ca87fa01-e4ad-43e0-8d9f-016dd22496c9)
> ![image](https://github.com/user-attachments/assets/304062df-0138-422a-b256-b81606648364)
> 

ah true
